### DISCUSSION TITLE
live desktop powered by Stellarium

### DISCUSSION THREAD

**Discussion by alephpi** [Stellarium/stellarium#3112] [General]
Hi, as a fan of nightskies, I am curious about the possibility of making live desktop powered by Stellarium. The idea is simply render a live  wallpaper for desktop with real-time simulation of nightskies by Stellarium. How can we realize it? What's the possible techniques/toolkits it may need?

Any idea is welcome. 🤗

**Comment by gzotti** [comment]
From historical views into the old forum I think this may have been possible or was short before introduction in the decade of WindowsXP. It may have become more difficult (or even impossible) when Windows Vista or 7 was introduced. 
Or maybe you mean some other operating system? 

However, you may simply have Stellarium running in a screen-sized window, with your applications on top.

**Reply by alephpi** [reply]
Alright, what about Linux?

**Reply by gzotti** [reply]
Well, what are the options to make a live Linux desktop from any graphics program? E.g. a game demo? Or at least a simple Video screen?

**Reply by 10110111** [reply]
Well, with Linux the option of running a maximized window with decorations off and set to always remain below all windows also works. At least in KWin both options to remove decorations and keep the window below all are available from the window menu.

Of course, this won't let you see your desktop icons, but you might not want them anyway.

**Comment by 10110111** [comment]
Another way, if you don't need the frame rate to be high, and can be satisfied by updates every, say, 5 seconds, you can make a Stellarium script that would make a screenshot periodically and pass the file to some command that would set wallpaper for your desktop environment.

Ways to change wallpaper from command line are discussed [here](https://superuser.com/q/488232/87525) for KDE and [here](https://askubuntu.com/q/66914/113421) for Gnome Shell.

**Reply by alephpi** [reply]
Yes, definitely a practical solution. since I'm not going to staring at my screen idly. It will be just like a joyful clock (instead of pure numbers). And I think it's more memory-saving.

**Comment by gzotti** [comment]
Ah, eighteen years ago the first team had a bunch of requests and discussion, but no solution.

https://sourceforge.net/p/stellarium/discussion/278769/thread/c984ccb2/
https://sourceforge.net/p/stellarium/discussion/278769/thread/fbbb72c1/
https://sourceforge.net/p/stellarium/discussion/278769/thread/265f0cab/
https://sourceforge.net/p/stellarium/discussion/278769/thread/5c9f93c9/
https://sourceforge.net/p/stellarium/discussion/278769/thread/239c3d2f/
https://sourceforge.net/p/stellarium/discussion/278769/thread/055bf11c/

**Reply by alephpi** [reply]
I've checked out all the discussions. Seems screenshot is a doable way. I just wonder if it's possible to direct export/render an image from Stellarium with command line only (as asked [here](https://sourceforge.net/p/stellarium/discussion/278769/thread/fbbb72c1/#458c))?

**Reply by gzotti** [reply]
I don't know whether a OpenGL screenshot can be made without full initialisation of a graphics window. This basically would require a way to make, at program start, exactly one image after setting up everything and store that in a screenshot without even launching the program's main window. This is beyond intended use.
You may have a running process and periodically (cronjob) emit an HTTP command that triggers making a screenshot.
